test/identity_cache_test.rb

```diff
@@ -23,4 +27,16 @@ def test_should_use_cache_in_transaction
       assert_equal false, IdentityCache.should_use_cache?
     end
   end
+
+  def test_should_use_cache_inside_transaction_on_specific_model
+    ModelWithConnection.transaction do
+      assert_equal false, IdentityCache.should_use_cache?
+    end
+  end
+
+  def test_should_use_cache_inside_unjoinable_transaction
+    ActiveRecord::Base.transaction(joinable: false) do
+      assert_equal true, IdentityCache.should_use_cache?
+    end
+  end
```

Comment:
Could you add a test case that uses `self.use_transactional_fixtures = false` so that you can test that should_use_cache? works without a transaction (i.e. the NullTransaction)